### ISSUE TITLE
Problems with Cue Notes Create

### ISSUE THREAD

**Issue by asherber** [finale-lua/lua-scripts#429]
I wanted to report a couple of bugs I've seen recently, but when I fired up Finale with commit 04450 of the repo, I found a new one: After I select a name for the cue and before I get the dialog to assign to a destination staff, I get an error indicating that `UseFullRowSelect()` is nil. 

I went back to commit a1bdc, before this change was introduced, and I am now able to reproduce the two other bugs I noticed:

1. Sometimes the cue name is not created in the Cue Notes category. It shows up instead in Technique Text and is also not set to scale with attached note.
2. Sometimes the cue shows up in the wrong octave. I'm seeing this right now picking a piano staff as the source and a Bb trumpet staff as the destination.

I'm on Windows 10, Finale 25.
![cue notes](https://user-images.githubusercontent.com/5248041/216843406-cb305ca4-d1d9-4263-91f2-23cf94fd3c78.gif)

cc: @cv-on-hub

**Comment by cv-on-hub** [comment]
Yeah - don't use older versions or you're inviting errors! It is easier at my end if you note the version number of the script rather than the commit number. (it's on the 6th line of all my scripts).

If you have a full set of `src` files, including the `library` and `mixin` folders in your scripts folder, then you can test the attached script, otherwise you'll have to wait until it is _distributed_ in the repo. 

These cue_name expressions never scale with attached notes - a germinal decision to avoid expressions jumping around at different sizes when re-positioning. Certainly as of v0.78 I can't see any way that a NEW expression will be created under "Technique Text" unless there's something screwy with category definition numbering on that file. Also octave swapping on the "Paste" can only be due to uniquely different transposition settings on the source and destination staves.
[cue_notes_create.lua.zip](https://github.com/finale-lua/lua-scripts/files/10613815/cue_notes_create.lua.zip)

**Comment by asherber** [comment]
This version throws an error "attempt to index a boolean value" at line 115 -- which has the comment 'this line throws error "attempt to index a boolean value"'!

**Comment by cv-on-hub** [comment]
Sorry about that. I left in some _test_ code. Try this one.
[cue_notes_create.lua.zip](https://github.com/finale-lua/lua-scripts/files/10613897/cue_notes_create.lua.zip)

**Comment by asherber** [comment]
Thanks, that one works for me with no errors.

In the document I had been working in, which is only a week old and pretty clean, using the plugin still does create name expressions in Technique Text. In a brand-new document, they're created in the Cue Notes category. Is there any info I can provide to help debug this?

I'm also able to recreate the octave swap on a brand new document. New doc with setup wizard, choose one Blank staff, one Bb Trumpet staff. Enter music in the blank staff similar to the image above. Use plugin to create cue notes in trumpet staff, and the notes are an octave too low. (Blank staff has no transposition; trumpet staff is standard Bb up M2.)

Finally, thank you for adding the ability to specify multiple target staves! There's one visual artifact in the dialog now, but I'm not sure if you have control over it. The column with the instrument names and checkboxes is narrower than the parent control, so there's a little vertical line in the header; I can actually grab this to resize the column. I think what's wanted is to make sure the column is the same size as the parent, or maybe to specify that you can't resize columns, but I don't know if you can do that at this level of abstraction.

![Screenshot_1](https://user-images.githubusercontent.com/5248041/216848714-71982ade-aabb-4f78-8e6a-4d91ccb0d0d6.png)

**Comment by cv-on-hub** [comment]
Thanks for helping me nail the problem with expressions in the wrong category. Works differently on Mac but should now be OK on both. (The problem occurred when a "Cue Names" category already existed containing no expressions).

Octave swap: this should have precisely the same result as copying music from one staff to another. I can't duplicate the error or see any reason you would get a discrepancy if the staves are set up with the same octave transposition. 

I can't find a way to alter the visual anomaly on the data list under Lua. At least its benign.
[cue_notes_create.lua.zip](https://github.com/finale-lua/lua-scripts/files/10632794/cue_notes_create.lua.zip)

**Comment by asherber** [comment]
Arggh! The issue with the octave swap was that I've only recently started working in Finale 25 (I know, still a little behind the times), and the preferences setting "Respect instrument ranges when copying or changing instruments" was on. When you mentioned that it should be the same as copying music, somehow I remembered that setting. I have no idea why Finale defaults that to On, because it's the kind of thing that can drive you crazy if you don't know it's there (or forget about it).

Also, the particular measures that I was seeing this issue on are perfectly within range for the destination instrument (trumpet), so I don't know what Finale was thinking. But turning the option off got everything working.

Thanks for working through the other issues.
